modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(() -> AbstractResourceManager
+                .encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+        Mockito.when(userManager.getUser(id,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(userOld);
+        User validatedUser = (User) ServerSideValidator.validateUpdatedSCIMObject(userOld, userNew, schema);
+        Mockito.when(userManager.updateUser(any(User.class), anyMap())).thenReturn(validatedUser);
+        SCIMResponse scimResponse = userResourceManager.updateWithPUT(id, scimObjectString, null,
+                attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestUpdateWithPUTNotFoundException")
+    public Object[][] dataToTestUpdateWithPUTNotFoundException()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING, schema, new User());
+        String id = userOld.getId();
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_UPDATE, schema, new User());
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, "userName", null,
+                        userNew, userOld, ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestUpdateWithPUTNotFoundException")
+    public void testUpdateWithPUTNoUserExistsWithTheGivenUserName(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object scimNewUserObject,
+                                                                  Object scimOldUserObject,
+                                                                  int expectedScimResponseStatus)
+            throws BadRequestException, CharonException, NotFoundException, NotImplementedException {
+
+        User userNew = (User) scimNewUserObject;
+        User userOld = (User) scimOldUserObject;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+        Mockito.when(userManager.getUser(id,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(null);
```

Comment:
merge this into 1 line

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11